pkg/resources/streamlit_acceptance_test.go

```diff
@@ -0,0 +1,428 @@
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "show_output.0.schema_name", schemaId.Name()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "show_output.0.title", "foo"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "show_output.0.owner", snowflakeroles.Accountadmin.Name()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "show_output.0.comment", "foo"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "show_output.0.query_warehouse", warehouse.ID().Name()),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "show_output.0.url_id"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "show_output.0.owner_role_type", "ROLE"),
+
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.name", id.Name()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.title", "foo"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.root_location", rootLocationWithCatalog),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.main_file", "foo"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.query_warehouse", warehouse.ID().Name()),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.url_id"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.default_packages"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.user_packages.#"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.import_urls.#"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.external_access_integrations.#", "1"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.external_access_integrations.0", integrationId.Name()),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.external_access_secrets")),
+			},
+			// unset
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Streamlit/basic"),
+				ConfigVariables: m(false),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "name", id.Name()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "database", databaseId.Name()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "schema", schemaId.Name()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "stage", stage.ID().FullyQualifiedName()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "directory_location", ""),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "main_file", "foo")),
+			},
+		},
+	})
+}
+
+func TestAcc_Streamlit_complete(t *testing.T) {
+	databaseId := acc.TestClient().Ids.DatabaseId()
```

Comment:
This setup is repeated throught the tests, but its cluttered. Maybe it could be cleaned up or at least organized in more redable way (?)

Reply:
Cleaned up a bit.